server/service/labels.go

```diff
@@ -354,6 +378,15 @@ func (svc *Service) DeleteLabelByID(ctx context.Context, id uint) error {
 	if err != nil {
 		return err
 	}
+	if label.LabelType == fleet.LabelTypeBuiltIn {
+		return fleet.NewInvalidArgumentError("label_type", fmt.Sprintf("cannot delete built-in label '%s'", label.Name))
+	}
+	for name := range fleet.ReservedLabelNames() {
+		if label.Name == name {
+			return fleet.NewInvalidArgumentError("name", fmt.Sprintf("cannot delete built-in label '%s'", label.Name))
+		}
+	}
```

Comment:
@mna, it should be in most cases. I included it to the side of potentially overprotecting against edge cases but it may not be worth the cognitive overhead. My thinking was that datastore values can be changed by users. I thought it would be good to be explicitly enforcing reserved names in the code rather than just relying on the `builtin` flag.

One potential edge case that this does address is where a user (accidentally) modified one of the builtin labels to be non-builtin (which happens in the repro steps of the ticket). This would at least prevent that label from being subsequently deleted. 

Happy to explore alternate approaches if you think it would be better to key everything off the `builtin` flag

Reply:
> I thought it would be good to be explicitly enforcing reserved names in the code rather than just relying on the builtin flag.

Makes sense to me! No harm in being overly cautious here!